File: algorithms/0011.container_with_most_water/container_with_most_water.cpp
class Solution {
public:
    int maxArea(const vector<int>& heights) {
        int area = 0;
        int lo = 0, hi = heights.size() - 1;
        while (lo < hi) {
            const int height = min(heights[lo], heights[hi]);
            const int width = hi - lo;
            area = max(area, height * width);
            if (heights[lo] < heights[hi]) {
                ++lo;      
            } else {
                --hi;
            }
        }
        return area;
    }
};

File: algorithms/0011.container_with_most_water/container_with_most_water.rs
impl Solution {
    pub fn max_area(heights: Vec<i32>) -> i32 {
        let mut area = 0;
        let (mut lo, mut hi) = (0, heights.len() - 1);
        while lo < hi {
            let height = heights[lo].min(heights[hi]);
            let width = (hi - lo) as i32;
            area = area.max(height * width);
            if heights[lo] < heights[hi] {
                lo += 1;
            } else {
                hi -= 1;
            }
        }
        area
    }
}